dcos/cosmos.py

```diff
@@ -180,7 +180,8 @@ def _cosmos_request(self,
                         response.headers.get('Content-Type')))
             return response
         except DCOSBadRequest as e:
-            if len(headers_preference) > 1:
+            error_type = e.response.json().get('type')
+            if error_type == 'not_valid' and len(headers_preference) > 1:
```

Comment:
@jsancio How do you propose we make them consistent? Should we just change the names? Should we create CosmosErrors for each case and convert to a CosmosError? Does this not have the potential to break a lot of clients?